crates/sui-core/src/signature_mpc/identifiable_abort.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use dashmap::DashMap;
+use twopc_mpc::secp256k1::paillier::bulletproofs::PartialDecryptionProof;
+
+use mysten_metrics::spawn_monitored_task;
+use signature_mpc::twopc_mpc_protocols;
+use signature_mpc::twopc_mpc_protocols::{
+    generate_proof, identify_message_malicious_parties, AdditivelyHomomorphicDecryptionKeyShare,
+    DecryptionKeyShare, PaillierModulusSizedNumber, PartyID,
+    SignaturePartialDecryptionProofVerificationParty,
+};
+use sui_types::base_types::EpochId;
+use sui_types::messages_signature_mpc::{
+    SignMessage, SignatureMPCMessageProtocols, SignatureMPCMessageSummary, SignatureMPCSessionID,
+};
+
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::signature_mpc::sign::SignState;
+use crate::signature_mpc::submit_to_consensus::SubmitSignatureMPC;
+
+/// Generate a list of proofs, one for every message in the message batch that its decryption failed.
+/// Each proof proves that the executing party ID (state.party_id) while signing on that message.
+pub fn generate_proofs(
+    state: &SignState,
+    failed_messages_indices: &Vec<usize>,
+) -> Vec<(
+    PartialDecryptionProof,
+    SignaturePartialDecryptionProofVerificationParty,
+)> {
+    let decryption_key_share = DecryptionKeyShare::new(
+        state.party_id,
+        state.tiresias_key_share_decryption_key_share,
+        &state.tiresias_public_parameters,
+    )
+    .unwrap();
```

Comment:
Why do you use `unwrap()`? the general rule is to never use it.
I'm aware the some of the code is using it, but we should not introduce more problems.